pkg/cache/cache.go

```diff
@@ -145,6 +145,15 @@ type Options struct {
 	// instead of `reconcile.Result{}`.
 	SyncPeriod *time.Duration
 
+	// FailOnUnknownResource configures the cache to return a ErrResourceNotCached error when a user
+	// requests, using Get() and List(), a resource the cache does not already have an informer for.
+	//
+	// This error is distinct from an errors.NotFound.
+	//
+	// Defaults to false, which means that the cache will start a new informer
+	// for every new requested resource.
+	FailOnUnknownResource bool
```

Comment:
For me, it's not directly clear what an `UnknownResource` is.
Could we maybe rename this to `FailReadWithoutRegisteredInformer`?

Reply:
Maybe `ReaderFailOnMissingInformer`?

Reply:
Sounds good! Will update

Reply:
Done

Reply:
Does the GoDoc here clear it up sufficiently? We've gone back and forth on the names here a couple of times and I'm not sure we will find something perfect. If the doc makes it sufficiently clear I think it's probably a good stopping place.

Reply:
(disregard, thanks @sbueringer )